packages/trie/src/trie.ts

```diff
@@ -350,6 +352,37 @@ export class Trie {
     await WalkController.newWalk(onFound, this, root)
   }
 
+  walkTrieIterable = _walkTrie.bind(this)
+
+  /**
+   * Executes a callback for each node in the trie.
+   * @param onFound - callback to call when a node is found.
+   * @returns Resolves when finished walking trie.
+   */
+  async walkAllNodes(onFound: OnFound): Promise<void> {
+    for await (const { node, currentKey } of this.walkTrieIterable(this.root())) {
+      await onFound(node, currentKey)
+    }
+  }
+
+  /**
+   * Executes a callback for each value node in the trie.
+   * @param onFound - callback to call when a node is found.
+   * @returns Resolves when finished walking trie.
+   */
+  async walkAllValueNodes(onFound: OnFound): Promise<void> {
+    for await (const { node, currentKey } of this.walkTrieIterable(
+      this.root(),
```

Comment:
Is it possible to make this user-definable so that e.g. we could start pass in a branch root on a lower level to skip a bunch of upper-level walking?

Reply:
Yes -- this is one example of a method that can be built using `walkTrieIterable`
       
The walk iterator can be set up to start from any node.  However note that the `currentKey` value returned for each node will be rooted from the starting node.  so to string together a node's "full key", you would still need the nibbles from the path to the starting node somehow.